cdap-ui/app/cdap/components/PreviewData/DataView/TableContainer.tsx

```diff
@@ -116,29 +160,29 @@ const TableContainer: React.FC<IPreviewTableContainerProps> = ({
           })}
         >
           <h2 className={classes.h2Title}>{T.translate(`${I18N_PREFIX}.outputHeader`)}</h2>
-          {outputs.map(([tableKey, tableValue], j) => {
-            const outputHeaders = tableValue.schemaFields;
-            const outputRecords = tableValue.records;
-            return (
-              <div key={`output-table-${j}`} className={classes.tableContainer}>
-                {outputs.length > 1 ? (
-                  <Heading type={HeadingTypes.h3} label={tableKey} />
-                ) : (
-                  <div
-                    className={classnames({
-                      [classes.headerBumper]: outputs.length === 1 && inputs.length > 1,
-                    })}
-                  />
-                )}
-                <DataTable
-                  headers={outputHeaders}
-                  records={outputRecords}
-                  isInput={false}
-                  previewStatus={previewStatus}
-                />
-              </div>
-            );
-          })}
+          {showOutputTabs
```

Comment:
Nit: Is there an easy way to consolidate this loop with the input table loop?

Reply:
Good call! I've refactored to reduce the duplicate code here, and done the same for the RecordContainer as well.